src/ui-scroll.js

```diff
@@ -386,8 +397,11 @@ angular.module('ui.scroll', [])
           // We need the item bindings to be processed before we can do adjustments
           !$scope.$$phase && !$rootScope.$$phase && $scope.$digest();
```

Comment:
@priandsf This is the bottleneck of the new demo (http://localhost:5005/fixedRowHeight/fixedRowHeight.html). And this $digest can't be removed, it responses for manual data-binding and transforming a row template into final html with all the data and handlers. Running the demo, you'll see that removing this $digest breaks the demo app.

From the other hand, I didn't see any significant difference in the performance with and without rowHeight (if don't touch this $digest). Probably the demo is not good, and I would ask you to update it if you can. The demo must show a difference with and without rowHeight. But if this particular $digest is the only call that "can" improve situation, then I would say probably we are on the wrong way. I don't see a situation where this $digest can be removed.

So, please look at the demo, try to reproduce performance changing and let me know, what do you think. Thanks!

Reply:
So you're right, commenting $digest in `processUpdates() `was too aggressive. It worked in my case as we had another trigger for $digest. I suspect this was not good :-)
I split the behavior into `row-height` and `allow-visibility-watch`, as they target 2 different use cases. I added the latest to the fixed row demo page (thanks for that one!) and it seems to work properly. Our use case is very similar to what this page shows, with a repeat within the row. Good guess!!
For `resizeAndScrollHandler()`, why are we forcing a $digest in the first place? When it does not update the DOM (handled by `enqueueFetch`), then it should just be a simple native scroll, shouldn't it? It is still conditioned by `rowHeight`, but I'm not sure that this is right :-)
I pushed that code to my branch.

Reply:
@priandsf Answering to your question, I'm not sure about resizeAndScrollHandler $digest, I need some time to investigate it. But the question I would like to consider before is what we are going to improve? Browser reflow problem resulting from `element.height()` and maybe something else related to direct DOM manipulations OR $digest things? The answer is mostly important and requires demo-proof. I think we should not optimize things that really don't affect the performance. You know easier code could be more valuable than performant code, in case its performance bonus is negligible.

Reply:
Fair enough - So far, `resizeAndScrollHandler()` in my app takes ~10ms, with most of the time within $digest when I scroll up and down without loading data. This can be higher when the GC is triggered. This is on a macbook pro running the latest Chrome. I haven't tried on other browsers yet...
So it doesn't feel laggy with this configuration. Up to you if you want/can remove the $digest.
Overall, the optimizations really made it smoother, thanks a lot for all your efforts. This librar
y rocks! 
![scroll-digest](https://user-images.githubusercontent.com/36373169/59074953-129c0000-889c-11e9-9e93-c3e3cd2b49e9.png)

Reply:
@priandsf I have an answer on why do we need $digest in the end of `resizeAndScrollHandler`: it is needed for triggering topVisible and bottomVisible properties changing (after they are calculated by `calculateProperties` method) on the outer scope (`top-visible="..."`) and to the outer scope adapter (`adapter.topVisible`). And we have two tests covering some of the correspondent use cases ([this](https://github.com/angular-ui/ui-scroll/blob/v1.7.3/test/BasicTestsSpec.js#L591) and [this](https://github.com/angular-ui/ui-scroll/blob/v1.7.3/test/BasicTestsSpec.js#L630)): they fail when that $digest is removed. And it does not depend on rowHeight. So when that $digest is removed, the assignments itself do work (scope and adapter props are being updated properly), but `$watch` does not work. The tests I just mentioned are about it.

One of the possible solutions is to html attribute for disable top/bottomVisible properties calculation/propagation + maybe adapter methods to enable/disable this feature runtime, and even the adapter method to calculate top/bottomVisible data immediately. So, users that don't need top/bottomVisible data (or don't need it right now) can switch this functionality off.

Reply:
Thanks for the investigation, that makes sense. The flag will work for me, although it might complicate the developer experience as understanding the attribute requires a deep knowledge of the internals. Based on our performance result, it is fine for me if we postpone that change.
Are you willing to merge the other ones?

Reply:
@dhilt I believe I got a solution for the $digest: the adapter checks if the directive has any of these attributes (top-visible, ...) defined and publishes it as a public property. Then the ui-scroll checks this property to know if it has to trigger the $digest.
I pushed the change to my branch. The unit tests and the demo (topVisible+topVisibleAdapter) work properly with this change. Let me know what you think!